internal/context/context.go

```diff
@@ -50,3 +51,14 @@ func GetAttemptSkewContext(ctx context.Context) time.Duration {
 	x, _ := middleware.GetStackValue(ctx, clockSkew{}).(time.Duration)
 	return x
 }
+
+// SetS3ResolvedURI sets the URI as resolved by the EndpointResolverV2
+func SetS3ResolvedURI(ctx context.Context, value string) context.Context {
```

Comment:
If we _only_ need this in S3 right now then let's just have it unexported in `service/s3`. We're still committing to a pseudo-API contract by doing this, even though it's under internal/, if we ever wanted to expand it to other services we wouldn't be able to remove it because it could break cross-module compat.

Reply:
Added a new file where I moved these operations and made them package private